### PR TITLE
Dex: history loading from chain or graph

### PR DESCRIPTION
**Pull request by 0xalecks**

### CODE REVIEW

File: projects/dex-ui/src/wells/useWellHistory.tsx [hunk 1]
```diff
@@ -1,117 +1,55 @@
...
 
-  // TODO: Replace with toHuman(format) function
-  const tmpFormatDollars = (input: string) => `$${parseFloat(input).toFixed(0)}`;
+export type WellEvent = SwapEvent | AddEvent | RemoveEvent;
 
-  const handleSwapEvents = (queryResults: GetWellEventsQuery) => {
-    if (!queryResults.well?.swaps || queryResults.well.swaps.length < 1) {
-      return [];
-    }
+// TODO: Store in env var?
```
Comment (marshall-bean) [inline projects/dex-ui/src/wells/useWellHistory.tsx:37]: Store what in env var?

**Review by marshall-bean — CHANGES_REQUESTED** [review]
A couple questions

File: projects/sdk/src/lib/BeanstalkSDK.ts [hunk 2]
```diff
@@ -44,8 +44,8 @@ export class BeanstalkSDK {
   public source: DataSource;
   public subgraphUrl: string;
 
+  public readonly instanceId: string;
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/sdk/src/lib/BeanstalkSDK.ts:47]: to uniquely identify an sdk instance in a dep array
```
useEffect(something, [sdk.instanceId)
```

doing just sdk was failing

File: projects/dex-ui/src/wells/historyLoader.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+import { Well } from "@beanstalk/sdk/Wells";
+import isEqual from "lodash/isEqual";
+import { BigNumber } from "ethers";
+import { GetWellEventsDocument } from "src/generated/graphql";
+
+const HISTORY_DAYS = 7;
+const HISTORY_DAYS_AGO_BLOCK_TIMESTAMP = Math.floor(new Date(Date.now() - HISTORY_DAYS * 24 * 60 * 60 * 1000).getTime() / 1000);
+
+const loadFromChain = async (sdk: BeanstalkSDK, well: Well): Promise<any[]> => {
+  console.log("Loading history from blockchain");
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/dex-ui/src/wells/historyLoader.ts:14]: i'm currently working on a logger so these will go away

File: projects/dex-ui/src/wells/useWellHistory.tsx (hunk 1 above)
Comment (0xalecks) [reply projects/dex-ui/src/wells/useWellHistory.tsx:37]: removed in next pr

**Review by marshall-bean — APPROVED** [review]

**Pull request merged**